pyomo/contrib/parmest/parmest.py

```diff
@@ -921,6 +1020,116 @@ def theta_est(
+        assert isinstance(self.multistart_sampling_method, str)
+        assert isinstance(solver, str)
+        assert isinstance(return_values, list)
+        
+        if self.n_restarts > 1 and self.multistart_sampling_method is not None:
+            # Generate theta values using the sampling method
+            theta_vals, results_df = self._generate_initial_theta(
+                self.estimator_theta_names, self.initial_theta, self.n_restarts, self.multistart_sampling_method
+            )
+
+            # make empty list to store results
+            for i in range(self.n_restarts):
+            # for number of restarts, call the self._Q_opt method
+            # with the theta values generated using the _generalize_initial_theta method
+
+                # Call the _Q_opt method with the generated theta values
+                objectiveval, thetavals[i], variable_values = self._Q_opt(
+                    ThetaVals=theta_vals,
+                    solver=solver,
+                    return_values=return_values,
+                )
+
+                # Check if the solver terminated successfully
+                if variable_values.solver.termination_condition != pyo.TerminationCondition.optimal:
+                    # If not, set the objective value to NaN
+                    solver_termination = variable_values.solver.termination_condition
+                    solve_time = variable_values.solver.time
+                    thetavals = np.nan
+
+                else:
+
+                    # If the solver terminated successfully, set the objective value
+                    init_objectiveval = objectiveval
+                    final_objectiveval = variable_values.solver.objective()
+                    solver_termination = variable_values.solver.termination_condition
+                    solve_time = variable_values.solver.time
+
+                # Check if the objective value is better than the best objective value  
+                if final_objectiveval < best_objectiveval:
+                    best_objectiveval = objectiveval
+                    best_theta = thetavals
+                    
+                # Store the results in a list or DataFrame
+                # depending on the number of restarts
+                results_df.iloc[i, :-4] = theta_vals
+                results_df.iloc[i, -4] = init_objectiveval
+                results_df.iloc[i, -3] = objectiveval
+                results_df.iloc[i, -2] = variable_values.solver.termination_condition
+                results_df.iloc[i, -1] = variable_values.solver.time
+
+                # Add buffer to save the dataframe dynamically, if save_results is True
+                if save_results and (i + 1) % buffer == 0:
+                    mode = 'w' if i + 1 == buffer else 'a'
+                    header = i + 1 == buffer
+                    results_df.to_csv(
+                        f"multistart_results.csv", mode=mode, header=header, index=False
```

Comment:
The file name is a string that the user should provide. You can make the default "multistart_results.csv"